3-Authorization-II/protect-web-api/AppCreationScripts/README_STEPS.md

```diff
@@ -0,0 +1,471 @@
+1. Go to your pom.xml, it should now be updated with build information for deploying to Azure App Services
+
+#### Step 2: Publish your files (java-spring-webapp-auth)
+
+1. In the terminal window in the base directory of your Java Spring project and enter the following command:
+
+        mvn clean package azure-webapp:deploy
+
+1. The terminal log should display a successful build after a few minutes stating that the deployment was successful
+
+> :note: The maven plugin uses information from the subscription Id you provide to deploy to Azure App Services. Deployment is subject to the limtations of your subscription. Modify your pom.xml to meet your subscription limits.
+
+#### Publishing using Visual Studio
+
+#### Step 1: Create a new app on Azure App Service
+
+1. Install the Visual Studio Code extension [Azure App Service](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azureappservice).
+1. Open the Azure App Service Extension and navigate to the App Services tab located under Resources
+1. Right-click on the App Services tab and select "Create New Web App"
+1. Enter a globally unique name for your web app (e.g. `java-spring-webapp-auth`) and press enter. Make a note of this name. If you chose `spring-webapi-domain` for your app name, your app's domain name will be `https://java-spring-webapp-auth.azurewebsites.net`
+1. Select `Java 11` for your runtime stack.
+1. Select `Java SE (Embedded Web Server)` for your Java web server stack.
+1. If you are asked for an OS, choose `Linux`.
+1. Select `Basic(B1)` or any other option for your pricing tier.
+
+#### Step 2: Publish your files
+
+1. Right-click on your newly created web app and select "Deploy to Web App"
+1. Progress through the flow until deployment begins, a prompt should appear notifying you that deployment is in progress
+1. A prompt should appear after a few minutes stating that deployment is complete.
+#### Update the Azure AD app registration (java-spring-webapp-auth)
+
+1. Navigate back to to the [Azure portal](https://portal.azure.com).
+In the left-hand navigation pane, select the **Azure Active Directory** service, and then select **App registrations (Preview)**.
+1. In the resulting screen, select the `java-spring-webapp-auth` application.
+1. In the app's registration screen, select **Authentication** in the menu.
+   - In the **Redirect URIs** section, update the reply URLs to match the site URL of your Azure deployment. For example:
+      - `https://java-spring-webapp-auth.azurewebsites.net/login/oauth2/code/`
+
+#### Update authentication configuration parameters (java-spring-webapp-auth)
+
+1. In your IDE, locate the `java-spring-webapp-auth` project. Then, open `webapp\src\main\resources\application.yml`.
+2. Find the key for **redirect URI** and replace its value with the address of the web app you published, for example, [https://java-spring-webapp-auth.azurewebsites.net/redirect](https://java-spring-webapp-auth.azurewebsites.net/redirect).
```

Comment:
```suggestion
2. Find the key for **redirect URI** and replace its value with the address of the web app you published, for example, [https://java-spring-webapp-auth.azurewebsites.net/redirect](https://java-spring-webapp-auth.azurewebsites.net/redirect).
```

I would recommend modifying the example redirect uri here, since in your project configuration file, you enter only the domain without path (e.g. localhost:3000 vs localhost:3000/redirect) so this could confuse people